packages/bierzo-wallet/src/logic/transactions/types/BwRegisterUsernameTx/ui/TransactionRow/Details.tsx

```diff
@@ -0,0 +1,95 @@
+import { ChainAddressPair, RegisterUsernameTx } from "@iov/bns";
+import { makeStyles } from "@material-ui/core";
+import { Table, TableBody, TableCell, TableHead, TableRow, Theme } from "@material-ui/core";
+import { Block, Typography } from "medulas-react-components";
+import * as React from "react";
+
+import { ChainAddress } from "../../../../../../components/AddressesTable";
+import { getChainName } from "../../../../../../config";
+import { ProcessedTx } from "../../../../types/BwParser";
+
+const useStyles = makeStyles((theme: Theme) => ({
+  sectionName: {
+    overflowWrap: "break-word",
+  },
+  tableHeader: {
+    "& > th": {
+      fontSize: "1.4rem",
+      color: theme.palette.text.primary,
+      padding: `${theme.spacing(1)}px ${theme.spacing(4)}px ${theme.spacing(1)}px 0`,
+    },
+  },
+  tableBody: {
+    "& > td": {
+      fontSize: "1.4rem",
+      color: theme.palette.text.secondary,
+      paddingLeft: 0,
+      padding: `${theme.spacing(1)}px ${theme.spacing(4)}px ${theme.spacing(1)}px 0`,
+    },
+  },
+}));
+
+interface Props {
+  readonly tx: ProcessedTx<RegisterUsernameTx>;
+}
+
+const TxDetails = ({ tx }: Props): JSX.Element => {
+  const classes = useStyles();
+  const [addresses, setAddresses] = React.useState<ChainAddress[]>([]);
+
+  React.useEffect(() => {
+    async function processAddresses(addresses: readonly ChainAddressPair[]): Promise<void> {
+      const chainAddresses: ChainAddress[] = [];
+      for (const address of addresses) {
+        chainAddresses.push({
+          ...address,
+          chainName: await getChainName(address.chainId),
+        });
+      }
+      chainAddresses.sort((a: ChainAddress, b: ChainAddress) =>
+        a.chainName.localeCompare(b.chainName, undefined, { sensitivity: "base" }),
+      );
+
+      setAddresses(chainAddresses);
```

Comment:
When you extract the function, let it return `readonly ChainAddress[]` such that the `setAddresses` call remains in the component